library/core/src/num/int_macros.rs

```diff
@@ -1307,6 +1307,33 @@ macro_rules! int_impl {
             (a as Self, b)
         }
 
+        /// Calculates `self + rhs + carry` without the ability to overflow.
```

Comment:
"ability to overflow" sounds like overflow is desirable and the lack of overflow is a limitation of this method. Is that what you meant?

Reply:
I would say
```
Extended summation of `self + rhs + carry`. The booleans are interpreted as a single bit
integer of value 0 or 1. If overflow occurs, then the boolean in the tuple returns 1. This
output carry can be chained into the input carry of another carrying add, which allows
for arbitrarily large additions to be calculated.
```

Reply:
FYI: Since this PR is merged, this discussion should go in the tracking issue (#85532).

Reply:
Wait pardon me this has already been merged. I will take discussion elsewhere or open my own PR.